test/unit/view_model/active_record/controller_test.rb

```diff
@@ -183,9 +183,20 @@ def test_destroy_missing
 
   #### Controller for nested model
 
-  def test_nested_collection_index
+  def test_nested_collection_index_associated
```

Comment:
The test expectations for `:index` and `:index_associated` are identical. If `index_associated` didn't apply a filter by parent id, we'd never know from this test.